pkg/solana/chainwriter/chain_writer.go

```diff
@@ -0,0 +1,324 @@
+package chainwriter
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"github.com/gagliardetto/solana-go"
+
+	commoncodec "github.com/smartcontractkit/chainlink-common/pkg/codec"
+	"github.com/smartcontractkit/chainlink-common/pkg/codec/encodings/binary"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/client"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/codec"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/fees"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/txm"
+)
+
+const ServiceName = "SolanaChainWriter"
+
+type SolanaChainWriterService struct {
```

Comment:
why are we [doing this](https://github.com/smartcontractkit/chainlink/blob/498b0b8579ad52a8c394fe3cbf55d3a86a8e29a0/core/services/relay/evm/chain_writer.go#L27-L33) differently than EVM?

Reply:
Is there an advantage of doing it the same way as EVM? The current way I'm doing it allows for Solana ChainWriter specific methods (such as `ResolveLookupTables`, etc. to be tested outside of the package.

Reply:
Makes sense!